Core/Src/Channels/RocketChannel.cpp

```diff
@@ -1,374 +1,341 @@
+static constexpr uint16_t CHAMBER_PRESSURE_GOOD_COUNT_MIN = 100;
+static constexpr uint16_t FLIGHT_BURN_TIME = 4500;
+static constexpr uint16_t AUTO_CHECK_BAD_COUNT_MAX = 10;
+
+RocketChannel::RocketChannel(
+	uint8_t id, const ADCChannel &fuelPressureChannel, const ADCChannel &oxPressureChannel,
+	const ADCChannel &chamberPressureChannel, ServoChannel &fuelServoChannel,
+	ServoChannel &oxServoChannel, PIControlChannel &piControlChannel, PyroChannel &internalIgniter1Channel,
+	PyroChannel &internalIgniter2Channel, PyroChannel &ventValveChannel, uint32_t refreshDivider
+) :
+	AbstractChannel(CHANNEL_TYPE_ROCKET, id, refreshDivider),
+	fuelPressureChannel(fuelPressureChannel), oxPressureChannel(oxPressureChannel),
+	chamberPressureChannel(chamberPressureChannel), fuelServoChannel(fuelServoChannel),
+	oxServoChannel(oxServoChannel), piControlChannel(piControlChannel),
+	internalIgniter1Channel(internalIgniter1Channel), internalIgniter2Channel(internalIgniter2Channel),
+	ventValveChannel(ventValveChannel), state(RS_INIT), stateOverride(RS_UNCHANGED),
+	can(Can::instance(0))
 {
 }
 
-int RocketChannel::init()
-{
+int RocketChannel::init() {
 	return 0;
 }
 
-int RocketChannel::exec()
-{
+int RocketChannel::reset() {
+	state = RS_INIT;
+	stateOverride = RS_UNCHANGED;
+	timeLastTransition = 0;
+	sensor_slope = 0.01888275146;
+	sensor_offset = -15;
+	chamberPressureMin = 0;
+	chamberPressureLowCounter = 0;
+	chamberPressureGoodCounter = 0;
+	autoCheckBadCounter = 0;
+	return 0;
+}
+
+int RocketChannel::exec() {
 	uint64_t time = STRHAL_Systick_GetTick();
-	if ((time - timeLastSample) < EXEC_SAMPLE_TICKS)
+	if ((time - timeLastSample) < EXEC_SAMPLE_TICKS) {
 		return 0;
-
+	}
 	timeLastSample = time;
 
-	// Current State Logic - executes state logic, also returns new state if transition conditions are met
-	internalNextState = currentStateLogic(time);
-
-	ROCKET_STATE nextState = state;
+	uint64_t stateTime = time - timeLastTransition;
 
-	if (externalNextState != state)
-	{ // Prioritize external event - there has to be some kind of priority, because internal could be different to external -> external means CAN -> either Sequence or Abort
-		nextState = externalNextState;
-	}
-	else if (internalNextState != state)
-	{
-		externalNextState = internalNextState; // Incase an internal state change happens, the external state, which is from some previous change would block it, so it is updated here
-		nextState = internalNextState;
+	ROCKET_STATE newState;
+	if (stateOverride != RS_UNCHANGED) {
+		newState = stateOverride;
+		stateOverride = RS_UNCHANGED;
+	} else {
+		newState = nextState(time, stateTime);
 	}
 
-	// Next State Logic
-	if (nextState != state)
-	{
-		nextStateLogic(nextState, time);
+	if (newState != RS_UNCHANGED) {
+		stateExit(state, time);
+		timeLastTransition = time;
+		stateEnter(newState, time);
+		stateTime = time - timeLastTransition;
+		state = newState;
 	}
 
+	stateDo(state, time, stateTime);
+
```

Comment:
Can you add a comment explaining how the state transitions work? Especially with the stateOverride and newState?